### DISCUSSION TITLE
Looking Mapbox js alternative "icon-allow-overlap": true, "text-allow-overlap": false in react-map-gl

### DISCUSSION THREAD

**Discussion by singhamritpal49** [uber/react-map-gl#1392] [Q&A]
I am having an issue with react-map-gl that is not displaying icons and text ( see attached images) properly (overlapping). I found some old code that I used in Mapbox js, which displayed everything correctly. Now I am using react but cannot find something useful.

map.addLayer({
        "id": "markers",
        "type": "symbol",
        "source": "markers",
        "layout": {
            "icon-image": "{marker-symbol}",
            "text-field": "{title}",
            **"icon-allow-overlap": true,
            "text-allow-overlap": false,**}
})

![Screen Shot 2021-03-03 at 3 23 51 PM](https://user-images.githubusercontent.com/48744353/110348045-3b1bd580-7fff-11eb-9cc3-da2494ca15a0.png)

![Screen Shot 2021-03-03 at 2 33 41 PM](https://user-images.githubusercontent.com/48744353/110348165-57b80d80-7fff-11eb-9b72-2fd3d44e000e.png)

**Comment by Pessimistress** [comment]
[Layer](https://visgl.github.io/react-map-gl/docs/api-reference/layer)

**Reply by singhamritpal49** [reply]
I looked at Layer documentation but did not find any property that is related to icon-allow-overlap": true, "text-allow-overlap. 
Are these built into the Layer component?

**Reply by Pessimistress** [reply]
It is just a wrapper of Mapbox GL JS’s layer descriptor. As the documentation states:

> The props provided to this component should be conforming to the Mapbox layer specification.

**Reply by eylonshm** [reply]
Hey, did you succeed ?
I'm trying to implement icon-allow-overlap and have some problems with it.
**How did you implement icon-allow-overlap, where did you add this code-line?**
Thank you :)

**Reply by singhamritpal49** [reply]
In order to get it working, I had to use Layer instead of markers (Source and Layer), 
I looked through these: 
https://visgl.github.io/react-map-gl/docs/api-reference/source
https://visgl.github.io/react-map-gl/docs/api-reference/layer

<Source id="my-data" type="geojson" data={geojson}>
        <Layer />
      </Source>

In Layer,  you use icon-allow-overlap": true, "text-allow-overlap properties

Let me know if that helped
